### PR TITLE
Add support for `targetextension` in xcode4

### PR DESCRIPTION
**Pull request by tempura-sukiyaki**

### CODE REVIEW

File: modules/xcode/xcode_common.lua [hunk 1]
```diff
@@ -892,11 +892,15 @@
 			settings['EXECUTABLE_PREFIX'] = cfg.buildtarget.prefix
 		end
 
-		--[[if cfg.targetextension then
+		if cfg.kind ~= "ConsoleApp" and cfg.targetextension then
```
Comment (samsinsane) [inline modules/xcode/xcode_common.lua]: Is there any particular reason why a `ConsoleApp` can't specify a `targetextension`? `app.out` for example?

File: src/_premake_init.lua [hunk 2]
```diff
@@ -1655,6 +1655,9 @@
 
 	-- Add variations for other Posix-like systems.
 
+	filter { "system:MacOSX", "kind:WindowedApp" }
+		targetextension ".app"
```
Comment (samsinsane) [inline src/_premake_init.lua]: Could this also apply to `system:ios`?

File: src/base/config.lua [hunk 3]
```diff
@@ -49,10 +49,14 @@
 		local bundlepath = ""
 
 		if cfg.system == p.MACOSX and kind == p.WINDOWEDAPP then
```
Comment (samsinsane) [inline src/base/config.lua]: Could this also apply to `p.iOS`?

File: src/base/config.lua [hunk 4]
```diff
@@ -49,10 +49,14 @@
...
-			bundlename = basename .. ".app"
+			bundlename = basename .. extension
 			bundlepath = path.join(bundlename, "Contents/MacOS")
 		end
 
+		if cfg.system == p.MACOSX and kind == p.CONSOLEAPP then
```
Comment (samsinsane) [inline src/base/config.lua]: Same comment as before, is there any reason why we're being so hostile towards `ConsoleApp`?

File: modules/xcode/xcode_common.lua (hunk 1 above)
Comment (tempura-sukiyaki) [reply modules/xcode/xcode_common.lua]: In xcode there is practically no way to add extensions at ConsoleApp.
Even if you put values in `WRAPPER_EXTENTION` or` EXECUTABLE_EXTENSION`, the product has no extension.

Comment (samsinsane) [reply modules/xcode/xcode_common.lua]: Oh ok, interesting. Thanks for the clarification!

File: src/base/config.lua (hunk 4 above)
Comment (tempura-sukiyaki) [reply src/base/config.lua]: It was the same reason I wrote above.
However, it is not good as it fails the test in this part.

File: src/_premake_init.lua (hunk 2 above)
Comment (tempura-sukiyaki) [reply src/_premake_init.lua]: I did not specifically think about `system: ios`.
Probably it is better to apply.

File: src/base/config.lua (hunk 3 above)
Comment (tempura-sukiyaki) [reply src/base/config.lua]: I did not specifically think about `system: ios`.
Probably it is better to apply.

File: src/_premake_init.lua (hunk 2 above)
Comment (tempura-sukiyaki) [reply src/_premake_init.lua]: Even if `_TARGET_OS` is set to `ios`, the generated project seems to be correct, but should I modify something by myself?
(e.g. `fliter { "system:macosx or ios", "kind:WindowedApp" }`)

**Pull request closed without merging**